xlators/mgmt/glusterd/src/glusterd.c

```diff
@@ -1388,6 +1388,23 @@ is_downgrade(dict_t *options, gf_boolean_t *downgrade)
     return ret;
 }
 
+void
+glusterd_destroy_hostname_list(xlator_t *this)
+{
+    glusterd_conf_t *priv = this->private;
+    glusterd_hostname_t *hostname_obj = NULL;
+    glusterd_hostname_t *tmp = NULL;
+
+    cds_list_for_each_entry_safe(hostname_obj, tmp, &priv->hostnames,
+                                 hostname_list)
+    {
+         cds_list_del_init(&hostname_obj->hostname_list);
```

Comment:
nit: you are accessing the struct members in the reverse order. First free hostname_obj->hostname, then del the list.